src/ut/httpresolver_test.cpp

```diff
@@ -150,13 +114,63 @@ TEST_F(HttpResolverTest, AAAARecordResolution)
   std::vector<DnsRRecord*> records;
   records.push_back(ResolverUtils::aaaa("cpp-common-test.cw-ngv.com", 3600, "3::1"));
   _dnsresolver.add_to_cache("cpp-common-test.cw-ngv.com", ns_t_a, records);
-  EXPECT_EQ("[3::1]:8888;transport=TCP",
-            HttpRT(_httpresolver).set_host("cpp-common-test.cw-ngv.com").set_port(8888).resolve());
+
+  std::vector<AddrInfo> targets = resolve(1, "cpp-common-test.cw-ngv.com", 8888);
+
+  ASSERT_GT(targets.size(), 0);
+  EXPECT_EQ(targets[0], ip_to_addr_info("3::1", 8888, IPPROTO_TCP));
+}
+
+/// Tests that the default time to remain on the blacklist is greater than
+/// 29999ms. Combined with the following test, effectively checks that the
+/// default time to remain on the blacklist is 30s.
+TEST_F(HttpResolverTest, BlacklistTimeLowerBound)
+{
+  add_white_records(11);
+  _httpresolver.blacklist(ip_to_addr_info("3.0.0.0"));
+  cwtest_advance_time_ms(29999);
+
+  EXPECT_TRUE(is_black("3.0.0.0", 11, 15));
+}
+
+/// Tests that the default time to remain on the blacklist is less than 30001ms.
+/// Combined with the previous test, effectively checks that the default time to
+/// remain on the blacklist is 30s.
```

Comment:
I think this would be better as one comment before the group of 4 tests, rather than 4 comments that reference each other.